builtin/logical/database/path_roles.go

```diff
@@ -2,263 +2,606 @@ package database
+		if creationStmtsRaw, ok := data.GetOk("creation_statements"); ok {
+			role.Statements.Creation = creationStmtsRaw.([]string)
+		} else if operation == logical.CreateOperation {
+			role.Statements.Creation = data.Get("creation_statements").([]string)
 		}
 
-		role.Statements.Revocation = strutil.RemoveEmpty(role.Statements.Revocation)
+		if revocationStmtsRaw, ok := data.GetOk("revocation_statements"); ok {
+			role.Statements.Revocation = revocationStmtsRaw.([]string)
+		} else if operation == logical.CreateOperation {
+			role.Statements.Revocation = data.Get("revocation_statements").([]string)
+		}
 
-		// Store it
-		entry, err := logical.StorageEntryJSON("role/"+name, role)
-		if err != nil {
-			return nil, err
+		if rollbackStmtsRaw, ok := data.GetOk("rollback_statements"); ok {
+			role.Statements.Rollback = rollbackStmtsRaw.([]string)
+		} else if operation == logical.CreateOperation {
+			role.Statements.Rollback = data.Get("rollback_statements").([]string)
 		}
-		if err := req.Storage.Put(ctx, entry); err != nil {
-			return nil, err
+
+		if renewStmtsRaw, ok := data.GetOk("renew_statements"); ok {
+			role.Statements.Renewal = renewStmtsRaw.([]string)
+		} else if operation == logical.CreateOperation {
+			role.Statements.Renewal = data.Get("renew_statements").([]string)
 		}
 
-		return nil, nil
+		// Do not persist deprecated statements that are populated on role read
+		role.Statements.CreationStatements = ""
+		role.Statements.RevocationStatements = ""
+		role.Statements.RenewStatements = ""
+		role.Statements.RollbackStatements = ""
 	}
+
+	role.Statements.Revocation = strutil.RemoveEmpty(role.Statements.Revocation)
+
+	return nil
 }
 
 type roleEntry struct {
-	DBName     string              `json:"db_name"`
-	Statements dbplugin.Statements `json:"statements"`
-	DefaultTTL time.Duration       `json:"default_ttl"`
-	MaxTTL     time.Duration       `json:"max_ttl"`
+	DBName        string              `json:"db_name"`
+	Statements    dbplugin.Statements `json:"statements"`
+	DefaultTTL    time.Duration       `json:"default_ttl"`
+	MaxTTL        time.Duration       `json:"max_ttl"`
+	StaticAccount *staticAccount      `json:"static_account" mapstructure:"static_account"`
+}
+
+type staticAccount struct {
+	// Fields used in Static Accounts and automatic password rotation
+
+	// Username to create or assume management for static accounts
+	Username string `json:"username"`
+
+	// Password is the current password for static accounts. As an input, this is
+	// used/required when trying to assume management of an existing static
+	// account. Return this on credential request if it exists.
+	Password string `json:"password"`
+
+	// LastVaultRotation represents the last time Vault rotated the password
+	// PasswordLastSet represents the last time a manual password rotation was
+	// preformed, using the Vault endpoint
+	LastVaultRotation time.Time `json:"last_vault_rotation"`
+	PasswordLastSet   time.Time `json:"password_last_set"`
```

Comment:
Still reviewing, but i haven't seen this used yet and wasn't able to find it on the page. is it still in use?

Reply:
It's not; we did not implement what we originally discussed regarding this. Removed in 01dab21e5